### PR TITLE
Clean up issues list, add documentation to help with named variadic confusion.

### PR DESCRIPTION
**Pull request by AndrolGenhald**
3 issues have been reported (#5500, #5503, #5508) since #5455 was released less than 12 hours ago, it seems there is a great deal of confusion with the new handling of variadic arguments. This is likely because named arguments are still new to most people, and even if they're aware of named arguments they may not consider the interaction with variadics.

This should improve the documentation where issues will pop up due to a missing `@no-named-arguments` annotation.
I also cleaned up issues list to match the issues that actually exist.

### CODE REVIEW

File: docs/running_psalm/issues.md [hunk 1]
```diff
@@ -5,8 +5,6 @@
  - [ArgumentTypeCoercion](issues/ArgumentTypeCoercion.md)
  - [AssignmentToVoid](issues/AssignmentToVoid.md)
  - [CircularReference](issues/CircularReference.md)
- - [ComplexFunction](issues/ComplexFunction.md)
- - [ComplexMethod](issues/ComplexMethod.md)
```
Comment (weirdan) [inline docs/running_psalm/issues.md:9]: Both seem to be thrown.

**Review by weirdan — CHANGES_REQUESTED** [review]

File: docs/running_psalm/issues.md (hunk 1 above)
Comment (AndrolGenhald) [reply docs/running_psalm/issues.md:9]: Well...perhaps we need to add documentation for those.

**Comment by muglug** [comment]
Thanks!

**Comment by drupol** [comment]
Thanks !

**Pull request merged**